packages/components/src/navigation/styles/navigation-styles.js

```diff
@@ -119,7 +120,6 @@ export const ItemBadgeUI = styled.span`
 	${ reduceMotion( 'animation' ) };
 `;
 
-export const ItemTitleUI = styled( Text )`
-	margin-right: auto;
-	text-align: left;
-`;
```

Comment:
I'd imagine that if we used SCSS directly, we'd have gained automatic RTL versions. 🤔

Reply:
Too bad styled-components doesn't have RTL support. I found a snippet that seems to work though: https://codesandbox.io/s/7w2pk6zz1x?file=/src/index.js

It uses https://github.com/cssjanus/cssjanus under the hood

Maybe worth experimenting with it?

Reply:
Possibly?
I haven't followed the development of G2 and styled components in Gutenberg, so I'm not sure!
Let's involve @ItsJonQ!

Reply:
We either use something automatic or develop our own little utils. Like we could have `{padding('1px 2px 3px 4px')}` function that properly replaces the values if document is in RTL mode.

Reply:
Great points! We used the pattern we saw here `packages/components/src/input-control/styles/input-control-styles.js:275` as an example, but I'm on board for standardization (if possible)

I'd love to hear Jon Q's input as well.

Reply:
Haii!!

> Too bad styled-components doesn't have RTL support

@david-szabo97 You are correct! There isn't a standardized way to handle RTL within CSS-in-JS yet.
There are several strategies, but the implementation differs between libraries.

(Will share more insight below, but first...)

> We either use something automatic or develop our own little utils."

We have one :)

https://github.com/WordPress/gutenberg/blob/master/packages/components/src/utils/rtl.js
It's a **very naive** (and tedious) way to do it.

I would love for that process to be automated.

### ☀️ Thoughts

For run-time compiled styles (e.g. Emotion/Styled components), you're able to tap into the inner CSS compiler ([Stylis](https://stylis.js.org/)) and add custom plugins. This "middleware" layer allows you to transform CSS styles just before they're rendered.

Naturally, this is where folks would add plugins to handle things like RTL (which is what [G2 Components does](https://github.com/ItsJonQ/g2/blob/master/packages/create-styles/src/createCompiler/plugins/index.js#L14)).

There's a couple more details to this, which I'll add in the ending notes:

The go-to RTL (JS) library is [CSSJanus](https://github.com/cssjanus/cssjanus). Unfortunately, it is licensed under Apache 2.0, which is **currently not compatible with Gutenberg's license**.

(That means that G2's RTL implementation will need to be removed if/when integrated into Gutenberg)

My regex skills aren't strong enough to create my own RTL solution. (I tried).

Hopefully this helps!

### 📝 Notes

* RTL in CSS-in-JS can be achieved by tapping into the plugin/middleware layer for Emotion
* A stylis RTL plugin can be added
* This needs to be added to a **custom Emotion cache**
* For Gutenberg's usage of emotion, (using `styled`), this can be achieved by creating and using an Emotion cache provider. All styled components would render inside. Something [like this](https://github.com/ItsJonQ/styled-providers/blob/master/src/emotion/ScopeProvider.js#L7).

Bonus: G2 Components can achieve this **without** a provider due to it's [custom Emotion implementation](https://github.com/ItsJonQ/g2/blob/master/packages/create-styles/src/createStyleSystem/createStyleSystem.js)

Reply:
Thanks for the insight here Jon! Much appreciated.